sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -138,6 +138,10 @@ trait ExpressionEvalHelper extends GeneratorDrivenPropertyChecks with PlanTestBa
       case (result: Float, expected: Float) =>
         if (expected.isNaN) result.isNaN else expected == result
       case (result: Row, expected: InternalRow) => result.toSeq == expected.toSeq(result.schema)
+      case (result: Seq[InternalRow], expected: Seq[InternalRow]) =>
```

Comment:
This change was needed because we can't do direct `equals()` comparison between UnsafeRow and other row classes. After this PR's changes, the _"SPARK-14793: split wide struct creation into blocks due to JVM code size limit"_ test case in `CodeGenerationSuite` was failing because the new code was producing `UnsafeRow` but the test code was comparing against `GenericInternalRow`. In the old code, this comparison between sequences of rows was happening in the default `case _ =>` below, but that case doesn't work when the `InternalRow` implementations are mismatched.

I'm not sure whether this change-of-internal-row-format will have adverse consequences in other parts of the code.